src/IROperator.h

```diff
@@ -1412,6 +1412,7 @@ inline Expr reinterpret(Expr e) {
  * argument. */
 inline Expr operator&(Expr x, Expr y) {
     user_assert(x.defined() && y.defined()) << "bitwise and of undefined Expr\n";
+    user_assert(!x.type().is_float() && !y.type().is_float()) << "bitwise and only takes integers\n";
```

Comment:
For uniformity with the other error messages, I'd change this to:

user_assert(x.type().is_int() || x.type().is_uint()) << "The first argument to bitwise and must be an integer or unsigned integer";
user_assert(y.type().is_int() || y.type().is_uint()) << "The second argument to bitwise and must be an integer or unsigned integer";